src/m365/spfx/commands/project/project-upgrade/rules/FN025001_ESLINTRCJS_overrides.ts

```diff
@@ -0,0 +1,432 @@
+            // STANDARDIZED BY:   eslint\\conf\\eslint-recommended.js
+            'use-isnan': 2,
+            // STANDARDIZED BY:   eslint\\conf\\eslint-recommended.js
+            // Set to 1 (warning) or 2 (error) to enable.
+            // Rationale to disable: !!{}
+            'no-extra-boolean-cast': 0,
+            // ====================================================================
+            // @microsoft/eslint-plugin-spfx
+            // ====================================================================
+            '@microsoft/spfx/import-requires-chunk-name': 1,
+            '@microsoft/spfx/no-require-ensure': 2,
+            '@microsoft/spfx/pair-react-dom-render-unmount': 1
+          }
+        },
+        {
+          // For unit tests, we can be a little bit less strict.  The settings below revise the
+          // defaults specified in the extended configurations, as well as above.
+          files: [
+            // Test files
+            '*.test.ts',
+            '*.test.tsx',
+            '*.spec.ts',
+            '*.spec.tsx',
+
+            // Facebook convention
+            '**/__mocks__/*.ts',
+            '**/__mocks__/*.tsx',
+            '**/__tests__/*.ts',
+            '**/__tests__/*.tsx',
+
+            // Microsoft convention
+            '**/test/*.ts',
+            '**/test/*.tsx'
+          ],
+          rules: {
+            'no-new': 0,
+            'class-name': 0,
+            'export-name': 0,
+            forin: 0,
+            'label-position': 0,
+            'member-access': 2,
+            'no-arg': 0,
+            'no-console': 0,
+            'no-construct': 0,
+            'no-duplicate-variable': 2,
+            'no-eval': 0,
+            'no-function-expression': 2,
+            'no-internal-module': 2,
+            'no-shadowed-variable': 2,
+            'no-switch-case-fall-through': 2,
+            'no-unnecessary-semicolons': 2,
+            'no-unused-expression': 2,
+            'no-with-statement': 2,
+            semicolon: 2,
+            'trailing-comma': 0,
+            typedef: 0,
+            'typedef-whitespace': 0,
+            'use-named-parameter': 2,
+            'variable-name': 0,
+            whitespace: 0
+          }
+        }
+      ]
+    };`;
+  }
+
+  get resolutionType(): string {
+    return 'js';
+  }
+
+  get severity(): string {
+    return 'Required';
+  }
+
+  get file(): string {
+    return './.eslintrc.js';
+  }
+
+  visit(project: Project, findings: Finding[]): void {
+    if (!project.esLintRcJs) {
+      return;
+    }
+
+    const nodes: ts.Node[] | undefined = project.esLintRcJs.nodes;
+    if (!nodes) {
+      return;
+    }
+
+    const hasOverridesProperty = nodes
+      .filter(node => ts.isIdentifier(node))
+      .map(node => node as ts.Identifier)
+      .filter(i => i.text === 'overrides').length !== 0;
+
+    if (!hasOverridesProperty) {
+      this.addFinding(findings);
```

Comment:
Here, let's use `addOccurrence` in conjunction with `addFindingWithOccurrences` so that we can capture the location in the file where changes should be added. This allows us to the location in the codetour report rather than top of the file.

Reply:
Nice. Updated.

Reply:
I've updated this so that if there is a finding, we return the position of `module.exports` line, as we are checking for whether `overrides` property exists to add the finding we can return a position for something that doesn't exist 😊